macros/abilities.cfg

```diff
@@ -103,11 +103,77 @@
     [/set_menu_item]
 #enddef
 
+#define METAMORPH_FIRE_NO_AIR ELEM TRANSFORM_MESSAGE IMAGE
+    [set_menu_item]
+        id=metamorph_{ELEM}
+        description={TRANSFORM_MESSAGE}
+        image={IMAGE}~SCALE(20,20)
+        [show_if]
+            [have_unit]
+                [not]
+                    type=EoMa_{ELEM}_Elemental,EoMa_{ELEM}_Avatar,EoMa_{ELEM}_God
+                [/not]
+                x,y=$x1,$y1
+                [filter_wml]
+                    [status]
+                        amulet_elem=yes
+                    [/status]
+                [/filter_wml]
+                [not]
+                    [filter_wml]
+                        moves=0
+                    [/filter_wml]
+                [/not]
+            [/have_unit]
+            [have_location]
```

Comment:
This one needs `x,y=$x1,$y1` 
Without it the location filter checks the whole map. Apart from that the code looks good. I've just tested it and it works (with the `x,y` fix).

Reply:
Oh, thanks